Userland/Libraries/LibCore/EventLoop.cpp

```diff
@@ -530,6 +543,11 @@ void EventLoop::wake_once(Object& receiver, int custom_event_type)
         post_event(receiver, make<CustomEvent>(custom_event_type), ShouldWake::Yes);
 }
 
+void EventLoop::add_job(NonnullRefPtr<Promise<NonnullRefPtr<Object>>> job_promise)
+{
+    m_running_jobs.append(move(job_promise));
```

Comment:
A Promise knows that it's been resolved or canceled, so how does it communicate to this class that it can be removed? It seems wasteful to leave the promises hanging around until the event loop is destroyed, which appears to be the way these are popped off? Or at least, until an event loop is removed from the stack of current event loops. How does this jobs vector get cleaned up to avoid accumulating garbage?

Reply:
It gets cleaned up when the event loop runs. (I will run at least once before exiting.)

Reply:
What do you mean? The only place you call remove() or any such API on the running jobs is in `~EventLoopPusher()`. That is only called when `exec()` and `spin_until()` are called, not during pump().

Reply:
Programming is hard